replica/memtable.cc

```diff
@@ -31,30 +31,26 @@ static mutation_reader make_partition_snapshot_flat_reader_from_snp_schema(
         std::any pointer_to_container,
         streamed_mutation::forwarding fwd, memtable& memtable);
 
-void memtable::memtable_encoding_stats_collector::update_timestamp(api::timestamp_type ts) noexcept {
-    if (ts != api::missing_timestamp) {
-        encoding_stats_collector::update_timestamp(ts);
-        min_max_timestamp.update(ts);
-    }
-}
-
 memtable::memtable_encoding_stats_collector::memtable_encoding_stats_collector() noexcept
     : min_max_timestamp(0, 0)
+    , min_live_timestamp(api::max_timestamp)
+    , min_live_row_marker_timestamp(api::max_timestamp)
```

Comment:
0 timestamp are indeed unlikely as typically they are derived from the wall clock as microseconds since unix epoch.
That said, application provided timestamp (and tests) may use 0 timestamps, so if it's the first timestamp tracked, the next timestamp will override it, even if the Comparator would have decided it shouldn't.

Reply:
> I suppose it is too late to change it now, considering that this information is written into the sstable metadata?

It can be fixed from now on.
The metadata collected in the sstables is indeed immutable, but there's extremely low chance the trackers lost a 0 value.